File: backend/app/schemas/node_types.py
"""NodeType schemas."""

from typing import Any

from pydantic import BaseModel, Field


class NodeTypeOut(BaseModel):
    name: str
    label: str = ""
    description: str = ""
    metadata: dict[str, Any] = Field(default_factory=dict)
    input_schema: dict[str, Any] | None = None
    output_schema: dict[str, Any] | None = None

File: frontend/src/api/nodeTypes.ts
import { api } from './request'

export interface JsonSchema {
  type?: string
  title?: string
  description?: string
  properties?: Record<string, JsonSchema>
  required?: string[]
  items?: JsonSchema
  anyOf?: JsonSchema[]
  $ref?: string
  $defs?: Record<string, JsonSchema>
  default?: unknown
}

export interface NodeTypeInfo {
  name: string
  kind: 'function' | 'condition'
  label: string
  description: string
  metadata?: Record<string, any>
  input_schema?: JsonSchema | null
  output_schema?: JsonSchema | null
}

export function listNodeTypes(): Promise<NodeTypeInfo[]> {
  return api.get('/node-types')
}
